apache/zeppelin
# Notebook: HDFS as a backend storage (Read & Write Mode)

### What is this PR for?

Zeppelin currently supports many backends for storing notes through Apache Commons VFS. 
Apache Commons VFS supports HDFS in readonly mode. 
This PR makes HDFS a first class citizen by allowing users to load notes from / save notes to HDFS.
### What type of PR is it?

Improvement
### Todos
### What is the Jira issue?

https://issues.apache.org/jira/browse/ZEPPELIN-1515
### How should this be tested?

Update zeppelin.notebook.dir property to a value like hdfs://localhost:9000/tmp/notebook and the property zeppelin.notebook.storage to the value org.apache.zeppelin.notebook.repo.HdfsNotebookRepo

check that your notes are loaded from and stored to HDFS by listing notes using the command :
hdfs dfs -ls /tmp/notebook
### Screenshots (if appropriate)
### Questions:
- Does the licenses files need update? No
- Is there breaking changes for older versions? No
- Does this needs documentation? Yes
